ols/app/main.py

```diff
@@ -53,4 +54,60 @@ async def rest_api_counter(
     return response
 
 
+@app.middleware("")
+async def log_requests_responses(
+    request: Request, call_next: Callable[[Request], Awaitable[Response]]
+) -> Response:
+    """Middleware for logging of HTTP requests and responses, at debug level."""
+    if logger.isEnabledFor(logging.DEBUG):
```

Comment:
this section can be moved after the check on line 81 to make it more clear :)

Reply:
not sure about how long a log line can be, we can send both request and response in one log entry , or we send 2 log entries at very close times after we have the response. So that request and response can be close in log.

Reply:
request and response should be separate log entries, because OLS will do(and log) other things in between getting the request and sending the response, which we want to see in order.

Reply:
Can't do one entry because the response body is streamed. There are at least 3:
- request
- response headers
- response body
They are all correlated by client_ip:port so we don't have to manually hunt for which is which.